### ISSUE TITLE
MauiImage resource as stream missing

### ISSUE THREAD

**Issue by AHComp** [dotnet/maui#12556]
### Description

I need access to resources from type MauiImage. If i insert the resource as embedded resource it works see here.

![image](https://user-images.githubusercontent.com/35998144/211656482-b40825b1-b3a4-4fd3-a4cc-3ed926a76425.png)

Now i have the resource double in the project as MauiImage and as embedded resource.

How i can access the MauiImage resource to get a stream to create a IImage or direct an IImage?

### Public API Changes

It's only needed to have a API to access the resource over the App object as static Methode.

### Intended Use-Case

I want reduce the package size so it's not possible to add Icons or images double.

**Comment by mattleibow** [comment]
Using `<MauiImage>` does not guarantee that the images are in an app-readable format. On Android it is put in some special format that can only be loaded using Android APIs, iOS performs some proprietary compression and optimizations that actually make the image invalid for most image loaders.

In addition, the image you provide is scaled and transformed to suit platform needs. For example if you have an image of size 100x100, the app will have generated 32x32, 64x64, 56x56 and all that. So the image you provide is more of a starting point and the compil;er makes it work for the app.

To load images in the form that I think you are looking for, you are probably wanting to use `<MauiAsset>`. This does not image transformations at all, meaning the svg stays svg, but if you are loading png files then this will work - and it will avoid all the OS transformations. 

If you really want to use svg files in your app and have them loaded via a stream as a png image file for some reason, this is technically possible but not easy. This will involve loading images in the OS system and then "saving" that image to a memory stream. This is not great for performance and may not be the best option.

TL;DR: use `<MauiAsset>`

**Comment by AHComp** [comment]
This is a feature request which information you need?
In Xamarin it was possible to load resources as Image and stream so please add this Funktion also for maui.
I you Statement you say it's possible so please implement.
It's a simple Funktion to access a bytearray !

**Comment by mattleibow** [comment]
I was asking if you used MauiAsset could you load it? Also, this feature is not really possible as mentioned in my comment above.

**Comment by AHComp** [comment]
Yes I try with Maui Asset but the I can't use the picture with the image command how I can access the mauiasset with ImageSource and the Image Control ?

I have a png and need to use as Image and stream for a drawable.

**Comment by AHComp** [comment]
https://stackoverflow.com/questions/72139593/convert-maui-imagesource-to-png

```
public class AndroidService : MyService
    {
        public async void Convert(string filename, ImageSource img)
        {
            System.IO.Stream outputStream = null;
            var handler = new FileImageSourceHandler(); 
            Bitmap pic = await handler.LoadImageAsync(img, Android.App.Application.Context);
            var savedImageFilename = System.IO.Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), filename);
            bool success = false;
            using (outputStream = new System.IO.FileStream(savedImageFilename, System.IO.FileMode.Create))
            {
                if (System.IO.Path.GetExtension(filename).ToLower() == ".png")
                {
                    success = await pic.CompressAsync(Bitmap.CompressFormat.Png, 100, outputStream);
                }
                else
                    success = await pic.CompressAsync(Bitmap.CompressFormat.Jpeg, 100, outputStream);
            }
        }
    }
```
For me this look not so difficult, is it possible that you have not so much experience with maui or xamarin ?

https://learn.microsoft.com/de-de/dotnet/maui/user-interface/controls/image?view=net-maui-7.0

```
Image image = new Image
{
    Source = ImageSource.FromFile("dotnet_bot.png")
};
```
So it should very easy to implement a Convert method or not ?

**Comment by mattleibow** [comment]
It is not entirely about the complexity, but more about performance. Your code is "correct", but is also not the "most correct". 

In your code you are decoding the image from the app (significant performance step) and then encoding it again (another significant step). When we load an image into an image view, we are not doing any loading but rather passing the resource ID to the control. This let's the OS do all the work and it is optimized specifically.

**Comment by mattleibow** [comment]
@AHComp I know the frustration on trying to get things to work, but you now mention the package size. This should not be an issue as we have been working on this for some time to get very small packages.

Can you open a new issue on that so we can see if we have some regressions? Or maybe you have already?

I will keep this issue open for the original request to have some shared way to load images from the `<MauiImage>` world.

I will try also get some code in a gist that can be used as well as try send a PR in to make this a part of the .NET MAUI SDK.

**Comment by mattleibow** [comment]
I think this will help: https://gist.github.com/mattleibow/c7c162e984d56557a72eb8edc4a459a0

It allows for loading images as a `Stream` without having to first go through the decode then encode then re-decode.

**Comment by Inrego** [comment]
How can it be done in .NET 8 (Windows)?
